src/System.Collections/tests/Generic/List/List.Generic.Tests.Misc.cs

```diff
@@ -202,11 +172,32 @@ public void InsertRangeValidations(T[] items, Func<T[], IEnumerable<T>> construc
                 Assert.Throws<ArgumentNullException>(() => list.InsertRange(0, null)); //"ArgumentNullException expected."
             }
 
-            public IEnumerable<T> ConstructTestCollection(T[] items)
+            public IEnumerable<T> ConstructTestEnumerable(T[] items)
             {
                 return items;
             }
 
+            public IEnumerable<T> ConstructLazyTestEnumerable(T[] items)
+            {
+                return ConstructTestEnumerable(items)
+                    .Select(item => item);
+            }
+
+            public IEnumerable<T> ConstructTestList(T[] items)
+            {
+                return items.ToList();
+            }
+
+            public Func<T[], IEnumerable<T>>[] GetCollectionGenerators()
+            {
+                return new Func<T[], IEnumerable<T>>[]
```

Comment:
Instead of creating the array every time why not just having a static array?

Reply:
@safern good thought - static won't work since we store an array of instance members, but I can initialize an array in the constructor so we only allocate the array once. Is that what you meant?

Reply:
Yes, I meant a previously initialized array and just use the same instance since it will always have the same values.

Reply:
:+1: I figured but I wanted to double check. I'll make that change, thanks!